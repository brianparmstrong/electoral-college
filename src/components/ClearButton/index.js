const ClearButton = () => {
  const clearStatus = () => {
    if (typeof(Storage) !== 'undefined') {
        localStorage.removeItem('ElectoralCollegeStatus');
        localStorage.removeItem('PopularVoteStatus');
        localStorage.removeItem('PopularVoteTotals');
        localStorage.removeItem('WinnerTakeAllTotals');
        alert('Cleared!');
        location.reload(); // eslint-disable-line no-restricted-globals
    } else {
      alert('Please enable Storage to save your selections.');
    }
  };

  return (
    <div className='buttonWrapper'>
      <button id='clearButton' type='button' onClick={clearStatus}>Clear</button>
    </div>
  );
};

export default ClearButton;
