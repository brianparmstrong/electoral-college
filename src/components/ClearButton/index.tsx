import { IfcClearButton } from '../../types';

const ClearButton = (clearButtonProps: IfcClearButton) => {
  const { handleClear } = clearButtonProps;

  return (
    <div className="buttonWrapper">
      <button id="clearButton" type="button" onClick={handleClear}>
        Clear
      </button>
    </div>
  );
};

export default ClearButton;
