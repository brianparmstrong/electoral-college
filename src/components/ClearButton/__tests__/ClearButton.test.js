import { render } from '@testing-library/react';
import ClearButton from '../';

it('renders correctly', () => {
  const { container } = render(<ClearButton />);
  expect(container.firstChild).toMatchSnapshot();
});
