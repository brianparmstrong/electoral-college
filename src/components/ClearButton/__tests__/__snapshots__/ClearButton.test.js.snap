// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="buttonWrapper"
>
  <button
    id="clearButton"
    type="button"
  >
    Clear
  </button>
</div>
`;
